### PR TITLE
feat(global-id): support clusters

### PR DESCRIPTION
**Pull request by mwegman**
- construct a hydra ID using a cluster
- get URL based on ID and cluster

# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* SDK Version
* Node/Browser Version
* NPM Version

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/common/src/uuid-utils.js [hunk 1]
```diff
@@ -1,27 +1,46 @@
 import {encode, decode} from './base64';
 import {SDK_EVENT, hydraTypes} from './constants';
 
-const hydraBaseResource = 'ciscospark://us';
 const hydraBaseUrl = 'https://api.ciscospark.com/v1';
 
+const isRequired = () => {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/common/src/uuid-utils.js]: This is a great way to handle required vars.

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 2]
```diff
@@ -32,14 +32,34 @@ const getConvoLimit = (options = {}) => {
 const Conversation = WebexPlugin.extend({
   namespace: 'Conversation',
 
+  /**
+   * @param {String} cluster the cluster containing the id
+   * @param {UUID} [id] the id of the conversation.
+   *  If empty, just return the base URL.
+   * @returns {String} url of the conversation
+   */
+  getUrlFromCluserId({cluster, id}) {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Was this meant to be `getUrlFromClusterId`?

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js [hunk 3]
```diff
@@ -393,10 +419,10 @@ const Conversation = WebexPlugin.extend({
    * provisional activity
    * @returns {Promise<Activity>}
    */
-  async leave(conversation, participant, activity) {
+  leave(conversation, participant, activity) {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: Glad we are able to remove the async modifiers to help clean up code readability.

File: packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js (hunk 2 above)
Comment (mwegman) [reply packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js]: You've never heard of a `cluser`? 😉
I will fix.

**Pull request merged**